rest/route/github.go

```diff
@@ -583,6 +589,102 @@ func (gh *githubHookApi) AddIntentForPR(pr *github.PullRequest, owner, calledBy
 		}
 	}
 
+	patches, err := patch.Find(patch.ByGithash(pr.Head.GetSHA()))
+	if err != nil {
+		grip.Info(message.Fields{
+			"message":           "error getting same hash patches",
+			"owner":             pr.Base.User.GetLogin(),
+			"repo":              pr.Base.Repo.GetName(),
+			"ref":               pr.Head.GetRef(),
+			"pr_num":            pr.GetNumber(),
+			"hash":              pr.Head.GetSHA(),
+			"override_existing": overrideExisting,
+			"err":               err.Error(),
+		})
+		return errors.Wrapf(err, "getting same hash patches")
+	}
+	// Remove a patch associated with this PR if applicable.
+	var filtered = []patch.Patch{}
+	for _, p := range patches {
+		if p.GithubPatchData.PRNumber != *pr.Number {
+			filtered = append(filtered, p)
+		}
+	}
+	patches = filtered
+	if len(patches) > 0 {
+		if !overrideExisting {
+			// We want to comment on explaining why a new patch will not be ran.
+			comment := fmt.Sprintf("There is an existing patch(s) for this commit SHA:\n%s\n\nPlease note that the status that is posted is not in the context of this PR but rather the (latest) existing patch and that may affect some tests that may depend on the particular PR. If your tests do not rely on any PR-specific values (like base or head branch name) than your tests will report the same status. If you would like a patch to run in the context of this PR and cancel the other(s), comment 'evergreen retry'.", gh.createPatchesWithPRLinks(patches))
+			grip.Info(message.Fields{
+				"message": "skipping CI on PR due to patch already existing",
+				"owner":   pr.Base.User.GetLogin(),
+				"repo":    pr.Base.Repo.GetName(),
+				"ref":     pr.Head.GetRef(),
+				"pr_num":  pr.GetNumber(),
+			})
+			return gh.sc.AddCommentToPR(ctx, pr.Base.User.GetLogin(), pr.Base.Repo.GetName(), pr.GetNumber(), comment)
+		} else {
+			// We want to cancel all existing patches and explain what happened on each one.
+			// After, we want to create a new patch (done below this scope).
+			grip.Info(message.Fields{
+				"message": "cancelling existing CI on same SHA patches",
+				"owner":   pr.Base.User.GetLogin(),
+				"repo":    pr.Base.Repo.GetName(),
+				"ref":     pr.Head.GetRef(),
+				"pr_num":  pr.GetNumber(),
+			})
+			for _, p := range patches {
+				err = model.CancelPatch(&p, task.AbortInfo{User: evergreen.GithubPatchUser, NewVersion: "", PRClosed: false})
+				if err != nil {
+					err = gh.sc.AddCommentToPR(ctx, pr.Base.User.GetLogin(), pr.Base.Repo.GetName(), pr.GetNumber(), "There was an issue aborting the other patches, please try 'evergreen retry' again.")
```

Comment:
This comment seems fine, but it may result in some confusion since it returns early and doesn't comment about the originating problem first (the overriding patch comment below this). Could we add this comment _after_ posting the comment with the context?